### ISSUE TITLE
Provide comparative size of masses on focus only - not while changing mass

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab#146]
Related to the implementation of `aria-valuetext` for the mass information (https://github.com/phetsims/gravity-force-lab/issues/129), I am wondering if it is possible to just provide the qualitative comparable mass size description upon focus, and not during changes to the mass sliders.

@zepumph, I am asking as I am not sure how dynamic we can make the arai-valuetext. I was not very explicit in the design document, though the examples in the design document for mass changes did not did not actually include the comparative mass description phrase. This phrase is only in the mass control focus examples.

These alerts are not currently part of the GFL BASICS sim and may not be needed there, so this issue may not be a high priority.

We can discuss briefly at our next meeting.

**Comment by zepumph** [comment]
I implemented this by having the slider update text to be the sparse version. Then changing the aria value text initially (on startup) to be the verbose one, and then setting the value text manually to the verbose version whenever the massControl is blurred (so that we get that when we come back to it). This isn't the most robust way of handling this, but I couldn't think of another way.

**Comment by zepumph** [comment]
@terracoda please review.

**Comment by terracoda** [comment]
@zepumph, this sounds good. There is one problem, however. The starting (focus) described size on the second mass sphere can be wrong after a change to the first and then a focus switch.

I think this will be tricky because the actual mass value in the `aria-valuetext` of the second mass sphere hasn't actually changed, only it's size in comparison to the other sphere has changed.

@zepumph, do you think there is anyway to the values of the two strings (actual mass and compared size) be managed separately? This seems unlikely to me. 

It would means that the `aria-valuetext` describing the actual size (size in kilograms) is managed by direct changes of the same mass slider, and then described compared size is managed by watching changes to the model that either mass slider can make. It would mean that the described size on both masses would have to be updated from the model like a dynamic description.

If this is possible, this could be useful in general. However, if this seem really hard right now, we might want to talk with others before trying it out.

**To reproduce the current issue:**
1. Move focus to mass 1 slider
2. Change the size of mass 1 to a lot bigger or a lot smaller than mass 2
3. Move focus to mass 2, and you will hear a comparable size that is out of synch with Mass 1.
4. Tabbing away from Mass 2 slider, again, updates the comparable size to the correct described comparison value.

**Here are four screen shots of the current issue from the A11y View:**
1. Mass 1 on focus - all is good.
<img width="756" alt="mass1-on-focus-both-values-good" src="https://user-images.githubusercontent.com/1902116/52329200-59631180-29f2-11e9-986b-47f75ffc2a9f.png">

2. Mass 2 slider focuses after a change in mass to mass 1 sphere - not good.
<img width="731" alt="mass2-focused-after-mass1-changed-out-of-synch" src="https://user-images.githubusercontent.com/1902116/52329308-b1017d00-29f2-11e9-9e1c-79c3529bb9ac.png">

3. Refocus on Mass 1 after no change in mass 2 - things go back into synch.
<img width="731" alt="mass1-refocus-all-back-in-synch" src="https://user-images.githubusercontent.com/1902116/52329367-e73efc80-29f2-11e9-872c-a00d57866e83.png">

4. But if there is a change in mass to mass 2, mass 1 will be out of synch on refocus.
<img width="731" alt="screen shot 2019-02-06 at 9 22 23 am" src="https://user-images.githubusercontent.com/1902116/52329420-0178da80-29f3-11e9-895b-977ce21264cd.png">

**Comment by zepumph** [comment]
@terracoda and I discussed perhaps looking into constantly updating the value text from the model instead of on blur. Nice edge case you found @terracoda! Sorry I didn't handle it.

**Comment by terracoda** [comment]
@zepumph, you are doing awesome work. I think your are right that handling the aria-valuetext from the model will be good approach.

**Comment by zepumph** [comment]
this should be fixed in REGULAR now. @terracoda can you point me in the right direction for BASICS? I'm not sure what exactly I'm supposed to be doing there for this issue. I don't think anything in this issue so far effects BASICS.

**Comment by terracoda** [comment]
@zepumph, the comparable mass descriptions are remaining in synch now.

Apologies for lack of clarity for how this issue relates to the BASICS sim.

Literal size descriptions (tiny, very small, small, etc) were removed from BASICS because of the huge scales in BASICS. Something that has a mass of 1 billion kg can't really be described as tiny.

However, comparative sizes may still be useful, especially if the comparison is are only provided on focus with dynamic `aria-valuetext`. 

I'll close this issue, and open a new one in BASICS repo with clearer direction. You may not have time to implement comparative descriptions for tomorrow's interview and that's ok.
